packages/runtime-common/helpers/indexer.ts

```diff
@@ -160,14 +160,45 @@ export async function setupIndex(
       );
       row.type = row.type ?? 'instance';
       row.last_modified = String(row.last_modified ?? now);
-      return asExpressions(
-        { ...defaultIndexEntry, ...row },
-        {
-          jsonFields: [...Object.entries(coerceTypes)]
-            .filter(([_, type]) => type === 'JSON')
-            .map(([column]) => column),
-        },
+
+      let columnNames = [
```

Comment:
Passing multiple entry types (instance, module, css) in `setupIndex` resulted in a SQL insert query where there were misaligned column names and values for different types, causing the query to return values in unpredictable order, making its return value array problematic to parse. Generating insert queries where all column are provided (even if undefined) fixes this issue. Hardcoding these columns like that could be difficult to maintain though... One alternative would be fetching them from the db but that would add overhead to tests, or maybe adding this list next to where `BoxelIndexTable` is defined so we don't forget to change this list when `BoxelIndexTable` gets updated

Reply:
this is likely to change over time. anythoughts on how we can keep this in sync with the schema? Perhaps we can perform a query to get all the column names from the table instead?

Here is an example where we get all the PK's from the tables in sqlite. https://github.com/cardstack/boxel/blob/main/packages/host/app/lib/sqlite-adapter.ts#L83-L97 Its not hard to imagine fashioning a query to get all the columns from a table from that. In pg, of source it is different, but quite straight forward. Perhaps the `DBAdapter` has a new method that returns the column names for a specified table that we implement in sqlite and pg, and use here so that we don't have to always update this when the schema changes.

Reply:
> yeah, you can set any column in the index as part of setup, so long as you are consistent and initialize the same columns for all the rows in your setup

Was trying to do that but then I realized it's a bit annoying setting consistent fields for entries of different types, that's why I resorted to a solution with deterministic insert queries

If you are ok with making this as a query on the adapter then great. Will adjust this in my next commit.

Reply:
yeah, i understand the frustration. i think a solution that allows developers to not have to worry about setting consistent fields is good. I think adding this as a query in the adapter is totally fine.